### ISSUE TITLE
reproducing the result on mini-MIAS

### ISSUE THREAD

**Issue by CarrotC** [brsegmentation/demonsSegmentation#7]
Thanks for your great work.
I ran the code to reproduce the result. But I got the mean of Hausdorff Distance as 34.24, much higher than what is mentioned in the paper. I tried twice and got the same results. Another difference is that the result I got shows the Number of clusters k is 4 in the first four folds and 6 in the last one, while k is 5 in the paper.
I just ran the main.m as specified in the README. Is there any other process I should do, or is there any bug in the code?
Looking forward to your reply.

**Comment by manish-k-sharma** [comment]
What matlab version and OS you used? Also, the source of ground truth and raw images is same as we used?
This is strange because the results were generated and verified multiple times by us.
There are some places where random intialization is required, but IIRR, we used fix seeds. If missed at some place, the seed may alter the results which is totally expected.

Can you just verify if all components of the data and processing requirements were same as we used?

**Comment by man-ks** [comment]
What matlab version and OS you used? Also, the source of ground truth and raw images is same as we used?
This is strange because the results were generated and verified multiple times by us.
There are some places where random intialization is required, but IIRR, we used fix seeds. If missed at some place, the seed may alter the results which is totally expected.

Can you just verify if all components of the data and processing requirements were same as we used?

**Comment by CarrotC** [comment]
I used matlab r2016a on a windows 10 machine. I just downloaded this project and ran main.m.
It seems that the ground truth masks mentioned in the paper have already in the directory data/MIAS/ground_truth and the raw images can be downloaded from the Internet automatically.
Is there anything else I should do? If none, I may try again using matlab r2014a though I don't think the version of matlab matters so much.

**Comment by CarrotC** [comment]
> What matlab version and OS you used? Also, the source of ground truth and raw images is same as we used?
> This is strange because the results were generated and verified multiple times by us.
> There are some places where random intialization is required, but IIRR, we used fix seeds. If missed at some place, the seed may alter the results which is totally expected.
> 
> Can you just verify if all components of the data and processing requirements were same as we used?

I used matlab r2016a on a windows 10 machine. I just downloaded this project and ran main.m.
It seems that the ground truth masks mentioned in the paper have already in the directory data/MIAS/ground_truth and the raw images can be downloaded from the Internet automatically.
Is there anything else I should do? If none, I may try again using matlab r2014a though I don't think the version of matlab matters so much.

**Comment by CarrotC** [comment]
I used matlab r2016a on a windows 10 machine. I just downloaded this project and ran main.m.
It seems that the ground truth masks mentioned in the paper have already in the directory data/MIAS/ground_truth and the raw images can be downloaded from the Internet automatically.
Is there anything else I should do? If none, I may try again using matlab r2014a though I don't think the version of matlab matters so much.

**Comment by man-ks** [comment]
Based on your description, your cluster formation is different than what we got, 4 clusters in four folds and 6 in one, while we got 5 in four folds and 6 in one. While identifying the initial number of clusters, we project to 2D using t-SNE. This requires initialization using random points. These random points can be dependent on seed. Depending upon this, the points can move around while clustering and hence the number of clusters can be different.

When the clusters are different, the atlas might differ and may result to poor segmentation.

**Comment by jasmainak** [comment]
@CarrotC slight differences are expected with different matlab versions and seeds but I wouldn't expect anything dramatically different. To help us, could you share the intermediate plots in Results/Mias/Plots? Perhaps the t-sne plots with the different clusters marked in different colors/symbols?

**Comment by CarrotC** [comment]
@jasmainak
[Plots.zip](https://github.com/brsegmentation/brsegmentation/files/4315825/Plots.zip)

**Comment by jasmainak** [comment]
thanks, your plots look good. I don't see an issue there. We have some more reporting scripts that generate html files and will help diagnose. @manish9396 will upload them to the repository by the end of this week and hopefully you can run them and share with us. 

In the meanwhile, could you tell us how you got the mean Hausdroff distance?

**Comment by CarrotC** [comment]
I got the mean Hausdroff distance by calculating the average value in Results/MIAS/hausdorffDistance.csv using excel.


[hausdorffDistance.zip](https://github.com/brsegmentation/brsegmentation/files/4322613/hausdorffDistance.zip)

**Comment by man-ks** [comment]
@CarrotC @jasmainak 
The results reported in the paper are in mm while the csv generated have the results measured from the pixels of the image where the resolution is 512*512.

Considering that the MIAS images are originally 1024*1024 and sampled at 200 micron pixel edge, the final Hausdorff Distance would be 
`hd = hd_pixel * 0.2 * 2`
where, 0.2 represents the 200 micron pixel edge and 2 is the compensation factor for downsampling.

If computed using the value reported in current thread, `34.24` will evaluate to `13.696` while our reported value is `13.34` which is fairly close.

I looked at the graphs too and these are similar to what we obtained and these look good.

Hope that answers your query.
